### ISSUE TITLE
How to store turnip prices long-term?

### ISSUE THREAD

**Issue by DerangedMind** [FrancisBourgouin/acnh-archipel#26]
While storing turnip prices in a linear array is a simple temporary solution, we will need to consider how to store the data long-term. 

Ideally, prices should be grouped by week. This could be a list of weeks or we could set keys for the week names (i.e., `${year}-${weekNumber}`). 

After the week ends, final trend name should be saved.

**Comment by FrancisBourgouin** [comment]
Personally, I think that the current storage solution for turnips is not meant to be temporary. 

Creating extra keys in the island object, or anywhere else is just redirecting the bloat of data somewhere else. 

I think the bigger question is what to do with all of the turnips prices once we have too much.

**Comment by DerangedMind** [comment]
> I think the bigger question is what to do with all of the turnips prices once we have too much.

I'm not sure that is a problem if it is organized properly. We should be able to support 8 years of entries without a problem, but let's say 1 year for simplicity. 

At 1 year of playing everyday, we have 52 weeks with 676 total entries, or 13 entries per week. 

Getting the latest entry is easy. Even this week is just a loop starting at the end and checking to see the dates are within the week and breaking the loop once we're done. What if I want to view a log of all the weeks I participated in and what my stats were? If I skipped a week or maybe a few days, we can't just divide by 13. What if I added the entries out of order? In a flat list, will that add more complexity to our search?

Alternatively, if the data is organized by week with a simple key name, we can generate the key based on the week of data we are looking for, get exactly what we need, and reduce on costly searches. We can also add metadata such as what the trend for that week was (which may become costly if we always have to generate it on the fly). 

Maybe this info isn't important and we just want to delete it each week, but I feel like we have been saying we want to hold onto it. 

What say you?
